include/ignition/gazebo/EntityComponentManager.hh

```diff
@@ -214,6 +214,16 @@ namespace ignition
       public: template<typename ComponentTypeT>
               ComponentTypeT *Component(const ComponentKey &_key);
 
+      /// \brief Get the data from a component.
+      /// * If the component type doesn't hold any data, this won't compile.
+      /// * If the entity doesn't have that component, it will return nullopt.
+      /// * If the entity has the component, return its data.
+      /// \param[in] _entity The entity.
+      /// \return The data of the component of the specified type assigned to
+      /// specified Entity, or nullptr if the component could not be found.
+      public: template<typename ComponentTypeT, typename DataType>
```

Comment:
Since we have `DataType` defined in our components as `Type`,  would it be possible to drop the second template argument?

I think we could do

```
template<typename ComponentTypeT>
std::optional<ComponentTypeT::Type> ComponentData(const Entity _entity) const;
```

Which I think makes calling cleaner, so you don't have to know the mapping from ComponentTypeT to DataType.

Reply:
Awesome, this is the piece of the puzzle that I was missing. Done in 88ad6319503e8f518a88827d02ba6b2e7b0bffa1